packages/transaction-controller/src/TransactionController.ts

```diff
@@ -944,6 +944,30 @@ export class TransactionController extends BaseController<
     this.update({ transactions: this.trimTransactionsForState(transactions) });
   }
 
+  /**
+   * Add params to transactionMeta.
+   *
+   * @param transactionMetaId - ID of the transaction.
+   * @param params - New params to be added to tarnsaction meta.
+   */
+  addTransactionMetaParams(
```

Comment:
We've intentionally not added a generic public update method to avoid breaking encapsulation, and to prevent any caller from manipulating the state without constraint and breaking the state and introducing instability.

Instead we provide targeted `updateX` methods such as `updatePreviousGasParams`, `updateTransactionSendFlowHistory`, and `updateTransactionGasFees`.

Each of these filters the inputs using `pickBy` to limit what properties can be updated, and also validates the status of the transaction being updated, typically `unapproved` only.

Can we replace this with a targeted `updateX` method and add suitable new properties to the `TransactionMetadata` type if needed?

Reply:
Exactly, I also want to add that giving this ability to make big updates could cause unpredicted issues since nothing here gets validated. This means any caller can update any transaction props which doesn't look ideal.

Reply:
+1, PR is updated to address this